webapi/Controllers/MaintenanceController.cs

```diff
@@ -46,14 +47,34 @@ public class MaintenanceController : ControllerBase
     [HttpGet]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
-    public ActionResult<MaintenanceResult?> GetMaintenanceStatus(
-        [FromServices] IKernel kernel,
+    public async Task<ActionResult<MaintenanceResult?>> GetMaintenanceStatusAsync(
+        [FromServices] IChatMigrationMonitor migrationMonitor,
         [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
         CancellationToken cancellationToken = default)
     {
+        MaintenanceResult? result = null;
+
+        var migrationStatus = await migrationMonitor.GetCurrentStatusAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
nit: ASP .net core doesnt have a SynchronizationContext so ConfigureAwait is not needed.

Reply:
Same comment for elsewhere in this PR.

Reply:
Ah great, nice to know!

Reply:
Updated